### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by alexlamsl**
Getting these out of the way before I have a go at #215 and #346 :sweat_smile:

### CODE REVIEW

**Comment by alexlamsl** [comment]
IMHO, we can merge `benchmark.conf` and `sample-cli-config-file.conf` to save ourselves a bit of maintenance overhead...

File: dist/htmlminifier.js [hunk 1]
```diff
@@ -848,7 +857,9 @@
       return attrValue;
     }
     else if (isMetaViewport(tag, attrs) && attrName === 'content') {
-      attrValue = attrValue.replace(/1\.0/g, '1').replace(/\s+/g, '');
+      attrValue = attrValue.replace(/\s+/g, '').replace(/[0-9]+\.[0-9]+/g, function(num) {
+        return (+num).toString();
```
Comment (kangax) [inline dist/htmlminifier.js]: Shouldn't `num` already be a string? This is redundant double-conversion, it seems.

File: dist/htmlminifier.js [hunk 2]
```diff
@@ -829,7 +838,7 @@
     }
     else if (isUriTypeAttribute(attrName, tag)) {
       attrValue = trimWhitespace(attrValue);
-      if (options.minifyURLs) {
+      if (options.minifyURLs && !isCanonicalURL(tag, attrs)) {
```
Comment (kangax) [inline dist/htmlminifier.js:849]: :+1:

File: dist/htmlminifier.js (hunk 1 above)
Comment (kangax) [inline dist/htmlminifier.js]: Oh, I see what you did there...

File: dist/htmlminifier.js [hunk 3]
```diff
@@ -1351,6 +1362,9 @@
         }
         if (options.minifyJS && isExecutableScript(currentTag, currentAttrs)) {
           text = minifyJS(text, options.minifyJS);
+          if (text.charAt(text.length - 1) === ';') {
+            text = text.slice(0, -1);
```
Comment (kangax) [inline dist/htmlminifier.js:1377]: Nice, yet another byte saved :)

File: dist/htmlminifier.js (hunk 1 above)
Comment (kangax) [inline dist/htmlminifier.js]: I would add a comment just to make it clear what's going on "0.9000" -> "0.9", "1.0" -> "1", "1.0001" -> "1.0001"

**Comment by kangax** [comment]
> IMHO, we can merge benchmark.conf and sample-cli-config-file.conf to save ourselves a bit of maintenance overhead...

Sure, go for it.

**Comment by alexlamsl** [comment]
variable renamed and comment added for meta-viewport optimisation

**Comment by kangax** [comment]
\o/

**Comment by kangax** [comment]
Couple nits for the future, `Object.create` not available in all environments but I guess it's been long enough and IE<9 is largely dead. I would also call `matchingMap` method `createMatchingMap` or `createMap`, but this is obviously a tiny nit.

File: dist/htmlminifier.js (hunk 3 above)
Comment (duncanbeevers) [inline dist/htmlminifier.js:1377]: Perhaps `text.match(/;$/)`?

Comment (alexlamsl) [inline dist/htmlminifier.js:1377]: That returns a whole match object, whereas `/;$/.test(text)` returns a boolean.

Anyway, I can't see how the RegExp version would give any performance advantage here.

**Pull request merged**